### PR TITLE
Make icon color optional

### PR DESCRIPTION
**Pull request by i-like-robots**
To avoid unexpected side effects with SVG images such as outlines being filled with colour.

The image service simply appends the following to SVG images when a colour is specified:

```
* { fill:*color* !important }
```

### CODE REVIEW

**Comment by keirog** [comment]
:+1:

File: tests/sass-setup.spec.js [hunk 1]
```diff
@@ -36,11 +36,16 @@ describe('next-sass-setup', function () {
...
+			expect(css).to.match(/\.next-icon\{[^}]+format=png&width=40/);
 		});
 
-		it('should output icon of correct colour', function () {
-			expect(css).to.match(/[^}]+(tint=%23FF0000,%23FF0000)[^}]+\1/);
+		it('should output icon of with specified color', function () {
+			expect(css).to.match(/.next-icon\{[^}]+(tint=%23FF0000,%23FF0000)[^}]+\1/);
+		});
+
+		it('can output icons without colors or dimensions', function () {
```
Comment (wheresrhys) [inline tests/sass-setup.spec.js]: dimensions are required for the ie8 fallback though, so I don't know if we should allow this

Comment (i-like-robots) [inline tests/sass-setup.spec.js]: It's only testing the SVG, I should re-word or re-organise the test case.

Comment (i-like-robots) [inline tests/sass-setup.spec.js]: @wheresrhys updated

**Comment by ironsidevsquincy** [comment]
This is useful, but all icons should handle setting a bg colour. Means designers have to supply icons with each object traced out, rather than using e.g. `ellipse` with `stroke`, as in https://github.com/Financial-Times/next-assets/blob/master/public-src/icons/org-people.svg?short_path=ac47c8d

**Comment by i-like-robots** [comment]
Do you think there could be an easy way to test for this?

**Comment by wheresrhys** [comment]
@i-like-robots A gotcha with delivering new icons via next-assets & image service - there is some very heavy caching so changes take ages (?forever) to come through

**Comment by i-like-robots** [comment]
I think forever... there may be a way to manually bust the cache.

**Comment by wheresrhys** [comment]
query string icon-thing.svg?v2. Can we make nextIcon() accept a version
parameter?


On 4 June 2015 at 15:05, Matt Hinchliffe <notifications@github.com> wrote:

> I think forever... there may be a way to manually bust the cache.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Financial-Times/next-sass-setup/pull/52#issuecomment-108908900>
> .
>

-- 

------------------------------
This email was sent by a company owned by Pearson plc, registered office at 
80 Strand, London WC2R 0RL.  Registered in England and Wales with company 
number 53723.

**Comment by ironsidevsquincy** [comment]
if the icons aren't out in the wild, you can purge the cache through fastly (though `next-geebee` doesn't seem to be respecting this)

otherwise, :+1: for rhys' version (ideally the image's url would have its hash in it, but can't think of an easy way of doing that)

**Comment by i-like-robots** [comment]
> if the icons aren't out in the wild, you can purge the cache through fastly 

Ahh cool, that'll do me for now.

**Pull request merged**